application/src/main/kotlin/net/corda/v5/application/messaging/FlowMessaging.kt

```diff
@@ -146,7 +146,7 @@ interface FlowMessaging {
      * @throws [CordaRuntimeException] if any session is closed or in a failed state.
      */
     @Suspendable
-    fun receiveAllMap(sessions: Map<FlowSession, Class<out Any>>): Map<FlowSession, Any>
+    fun receiveAllMap(sessions: Map<FlowSession, Class<out Any>>): Map<out FlowSession, Any>
```

Comment:
Should the out be added to the value not the key type?

Reply:
The bottom line is that our generic types are wrong, but we typically paper over such errors using `uncheckedCast()`. Unfortunately, `uncheckedCast()` is a kludge at best, and the problems it tries to hide frequently generate other warnings whenever we upgrade Kotlin.

At some point, we are going to trip over something that future Kotlin will just _refuse_ to compile :crystal_ball:.